lib/ansible/modules/async_wrapper.py

```diff
@@ -123,6 +124,15 @@ def _get_interpreter(module_path):
         module_fd.close()
 
 
+def _make_temp_dir(path):
+    # TODO: Add checks for permissions on path.
```

Comment:
not a blocker, but we might want to make 'makedirs_safe' a common function in module_utils (currently in lib.ansbile.utils.path) as its almost the same as this code.

Reply:
For now I'm just going to leave it. Making `makedirs_safe` actually safe would probably involve a whole other discussion in what constitutes safe when it comes to permissions. That should be done when the security concerns mentioned above are worked on. For now I just want to fix the race condition when the dir is created.